Enable WASM extensions by default

jwendell opened the pull request:
While working on enabling WASM by default (MAISTRA-2418), an issue was found (MAISTRA-2516).
This PR addresses both issues in separate commits.

Complement of https://github.com/maistra/istio-operator/pull/747

jwendell committed:
MAISTRA-2418: Add wasmExtensions section to global.yaml

So that it gets properly parsed.

Complement of https://github.com/maistra/istio-operator/pull/747

rcernich commented:
Does this need to include the fix for `isComponentEnabled()`?

jwendell commented:
> Does this need to include the fix for `isComponentEnabled()`?

I don't think so. It's a separate issue and deserve its own Jira ticket.

rcernich commented:
Maybe we change the issue to be a little broader in scope, e.g.: enable wasm extensions by default

jwendell commented:
The issue is already "enable wasm extensions by default". I just think that the issue we found yesterday should be handled separated as strictly speaking it's another issue that was just discovered because of this wasm issue.

rcernich commented:
it may be another issue, but without the fix, wasm extensions won't be enabled by default. :/

jwendell commented:
Correct, then we could hold this one and only merge it when the other issue is fixed

jwendell committed:
MAISTRA-2516: Consider values from global.yaml when rendering charts

Currently there's a bug in the code that renders charts that it's not
correctly merging the fields from global.yaml in the resulting chart.
The field status.appliedValues is also affected by this bug.

We fix the bug by setting back `spec.Istio` with the result of
the merge of its previous content and global.yaml.

rcernich commented on pkg/controller/versions/strategy_v2_1.go:
```diff
@@ -323,6 +323,9 @@ func (v *versionStrategyV2_1) Render(ctx context.Context, cr *common.ControllerR
 	}
 
+	// Update spec.Istio back with the previous content merged with global.yaml
+	spec.Istio = v1.NewHelmValues(values)
```
We shouldn't be updating the spec.  We should be updating appliedValues or just using the values as it was before.

rcernich replied:
Ignore what I said.  It looks like there were more issues than just the isComponentEnabled() invocation (which is correct with this change), with part of the code using the merged `values` and other parts using the user specified `spec.Istio`.  Also, I verified that `spec` is not the actual resource, but the results after applying profiles.
